htlcswitch/link.go

```diff
@@ -1014,11 +991,14 @@ out:
 			break out
 		}
 
-		// If the previous event resulted in a non-empty
-		// batch, reinstate the batch ticker so that it can be
-		// cleared.
-		if l.batchCounter > 0 {
+		// If the previous event resulted in a non-empty batch, resume
+		// the batch ticker so that it can be cleared. Otherwise pause
+		// the ticker to prevent waking up the htlcManager while the
+		// batch is empty.
+		if l.channel.PendingLocalUpdateCount() > 0 {
```

Comment:
this is certainly correct, but does this need to be executed on every iteration? not a blocker

Reply:
Maybe not, but it I put it there to be sure that the flow can never get stuck if we forget to set the timer somewhere. It may be a little defensive, but I think it is justified given the importance of this logic. I don't think there is significant performance impact of executing the check with every iteration.

Reply:
sounds good, agree with those comments